wifiphisher/common/victim.py

```diff
@@ -0,0 +1,159 @@
+        corresponding mac address
+
+        :param self: empty Victim instance
+        :type self: Victim
+        :param vmac_address: mac address of the victim
+        :type vmac_address: str
+        """
+
+        path_to_mac_file = constants.MAC_PREFIX_FILE
+        macmacther_instance = macmatcher(path_to_mac_file)
+        vendor = macmacther_instance.get_vendor_name(vmac_address)
+        Victims_instance = Victims.get_instance()
+        if vmac_address in Victims_instance.victims_dic:
+            Victims_instance.victims_dic[vmac_address].vendor = vendor
+        else:
+            raise Exception("Error: No such mac address exists in dictionary")
+
+    def assign_ip_to_victim(self, vmac_address, ip_address):
+        """
+        Updates the ip address of the victim with the corresponding mac address
+        """
+
+        Victims_instance = Victims.get_instance()
+        if vmac_address in Victims_instance.victims_dic:
+            Victims_instance.victims_dic[vmac_address].ip_address = ip_address
+        else:
+            raise Exception("Error: No such mac address exists in dictionary")
+
+
+class Victims():
+    """
+    Singleton class that manages all of the victims
+    """
+
+    # Instance will be stored here.
+    __instance = None
+
+    @staticmethod
+    def get_instance():
+        if Victims.__instance is None:
+            Victims()
+        return Victims.__instance
+
+    def __init__(self):
+        if Victims.__instance is not None:
+            raise Exception("Error: Victims class is a singleton!")
+        else:
+            Victims.__instance = self
+            self.victims_dic = {}
+
+    def add_to_victim_dic(self, Victim):
+        """
+        Add new victim to the dictionary
+        """
+
+        self.victims_dic[Victim.vmac_address] = Victim
+
+    def get_print_representation(self):
+        """
+        Returns the victim dictionary in str format in order to be printed
+
+        :param self: Victims instance
+        :type self: Victims
+        :rtype str
+        """
+
+        out = ""
+        for value in self.victims_dic.values():
+            out += value.vmac_address + '\t' + value.ip_address + '\t' \
+                     + value.os + '\t' + value.vendor + '\n'
+        return out
+
+    def associate_victim_ip_to_os(self, ip_address, url):
+        """
+        receives a victims ip address and request as input, finds the
+        corresponding os by reading the initial requests file and then accesses
+        the victim dictionary and changes the os for the victim with the
+        corresponding ip address
+
+        :param self: Victims instance
+        :type self: Victims
+        :param ip_address: ip address of the victim
+        :type ip_address: str
+        :param url: request of the victim
+        :type url: str
+        """
+
+        path_to_url_file = constants.URL_TO_OS_FILE
+        with open(path_to_url_file, "r") as url_file:
```

Comment:
Wouldn't be bether to have store the contents of the file in memory instead of reading it again and again?